subsys/bluetooth/host/adv.c

```diff
@@ -629,7 +640,7 @@ static int hci_set_ad_ext(struct bt_le_ext_adv *adv, uint16_t hci_op,
 		/* If possible, set all data at once.
 		 * This allows us to update advertising data while advertising.
 		 */
-		return hci_set_adv_ext_complete(adv, hci_op, ad, ad_len);
+		return hci_set_adv_ext_complete(adv, hci_op, total_len_bytes, ad, ad_len);
```

Comment:
This change is independent and should go in a separate commit. Maybe even a separate PR. It's easier for us reviewers when the PRs are kept small. (I know @Thalley asked for this in this PR, but personally I would prefer to review one change at a time.)

Reply:
I recommend using a separate commit instead of a separate pr.